### PR TITLE
Document GHC-56538

### PR DESCRIPTION
**Pull request by david-christiansen**
This one is tricky because the text of the error uses the term "head" in two different but related ways.

What do you think of my explanation?

### CODE REVIEW

File: message-index/messages/GHC-56538/index.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+title: Instance head is not headed by a class
+summary: A type class instance declaration is declared for something that is not a type class
+introduced: 9.6.1
+severity: error
+---
+
+Instance declarations specify the operations that place a type into a type class. For example, an instance of `Functor` describes how `fmap` works for a type. Instances only make sense for things that are, in fact, type classes - `instance Int Int` is nonsense, as is `instance thing Int`.
```
Comment (mknorps) [inline message-index/messages/GHC-56538/index.md:8]: Please add an example of `Ord` or `Eq`, as a Haskell newbie may have not encountered `Functor` type class, but usually how to overload equality is taught for other languages in CS 101.

File: message-index/messages/GHC-56538/index.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+A _head_ is the thing being applied at the beginning of a chain of applications. For instance, in `MonadReader Int MyMonad`, the head is `MonadReader`. When there are no arguments, the head is the entire construct, so `Int` is the head of `Int`. For an instance declaration to make sense, the head of the instance being defined must be the name of a type class. The terminology here is a bit overloaded, because the instance being defined is the _instance head_, and the head of the instance head must be a class.
+
+For example, in:
+```
+instance Show Int where
+  ...
+```
+the instance head is `Show Int` and its head is `Show`.
+
+Normally, this error is caught by a different step, where GHC uses the type system to ensure that the instance being defined is of a class. But a lower-case name is a type variable, which the type system cannot rule out.
```
Comment (mknorps) [inline message-index/messages/GHC-56538/index.md:19]: Could you provide a case of valid head that starts with a lowercase? Why the system cannot rule it out (or is it too deep ?).

Comment (elland) [reply message-index/messages/GHC-56538/index.md:19]: Or if it makes sense, cross-referencing documentation that explains this in more detail?

File: message-index/messages/GHC-56538/index.md (hunk 1 above)
Comment (david-christiansen) [reply message-index/messages/GHC-56538/index.md:8]: Great idea! Will do.

File: message-index/messages/GHC-56538/index.md (hunk 2 above)
Comment (david-christiansen) [reply message-index/messages/GHC-56538/index.md:19]: There are no valid ones that begin with a lowercase letter. I had to reverse-engineer what was going on in order to explain this - I don't know of better docs, unfortunately :-(

File: message-index/messages/GHC-56538/index.md (hunk 1 above)
Comment (david-christiansen) [reply message-index/messages/GHC-56538/index.md]: Done

**Pull request merged**